### PR TITLE
[Hsien Kit] iP

### PR DESCRIPTION
**Pull request by KHsienKit**

### CODE REVIEW

File: src/main/java/Deadline.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task {
+    private final String TYPE = "D";
+    private String by;
```
Comment (ZiliaAJY) [inline src/main/java/Deadline.java:12]: Consider declaring attributes as protected rather than private as it would be easier to facilitate extension of subclasses as the project progress.

File: src/main/java/V.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        System.out.println(text);
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    public static void greet() {
+        String logo = " _       _ \n"
+                + "\\ \\     / / \n"
+                + " \\ \\   / / \n"
+                + "  \\ \\_/ / \n"
+                + "   \\___/ \n";
```
Comment (ZiliaAJY) [inline src/main/java/V.java]: Consider to declare as public static final

File: src/main/java/V.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    public static void main(String[] args) {
+        greet();
+
+        boolean isOnline = true;
+        Task[] listOfTasks = new Task[100];
+        int count = 0;
+        String temp;
```
Comment (ZiliaAJY) [inline src/main/java/V.java]: variable name "temp" can be more descriptive.

File: src/main/java/V.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        Scanner input = new Scanner(System.in);
+        
+        while (isOnline) {
+            line = input.nextLine();
+            lineArr = line.trim().split(" ");
+            switch (lineArr[0]) {
+            case "bye":
+                input.close();
+                isOnline = false;
+                break;
```
Comment (ZiliaAJY) [inline src/main/java/V.java]: Good use of boolean flag isOnline as it allows the program to have a clear flow and makes the code easy to understand.

**Review by ZiliaAJY — COMMENTED** [review]
Good job!

File: src/main/java/V.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                System.out.println(listOfTasks[count]);
+                count++;
+                break;
+            default:
+                System.out.println("Try again");
+                break;
+            }
+        }
+        printBlock("See Ya");
+    }
```
Comment (shaunngoh) [inline src/main/java/V.java]: You could consider shortening your main method by breaking down the respective tasks into smaller methods

File: src/main/java/Event.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+public class Event extends Task {
+    private final String TYPE = "E";
```
Comment (shaunngoh) [inline src/main/java/Event.java:13]: Good use of declaring a constant to determine the type of task when returning from the getType method!

File: src/main/java/V.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        int count = 0;
+        String temp;
+        String description;
+        String by;
+        String[] timePeriod;
+        String from;
+        String to;
+        String line;
+        String[] lineArr;
+        Scanner input = new Scanner(System.in);
```
Comment (shaunngoh) [inline src/main/java/V.java]: The respective variables here are hard to follow, maybe you could provide a comment for each of the variable's use case?

File: src/main/java/V.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        
+        while (isOnline) {
```
Comment (shaunngoh) [inline src/main/java/V.java]: Good naming convention for your boolean variable!

**Review by shaunngoh — COMMENTED** [review]
Overall, I found that your code was easy to read, though maintainability might be slightly harder due to the complex nature of your main function. Goodjob!

File: src/main/java/Parser.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+     * Execute the function depending on the command and then returns the status of V.
+     * Return false if command "bye" was used, else returns true.
+     * @param command command from user
+     * @param isFromSaveFile true if command is read from a save file, else false
+     * @return status of V after a command
+     */
+    public boolean parse(String command, boolean isFromSaveFile) {
+        try {
+            wordArray = command.trim().split(" ");
+            switch (wordArray[0].toLowerCase()) {
```
Comment (okkhoy) [inline src/main/java/Parser.java:25]: Can SLAP this out.

File: src/main/java/Parser.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            case "mark":
+                position = Integer.parseInt(wordArray[1]);
+                taskList.markTask(position, isFromSaveFile);
+                return true;
+            case "delete":
+                position = Integer.parseInt(wordArray[1]);
+                taskList.deleteTask(position);
+                return true;
+            case "todo":
+                description = String.join(" ", Arrays.copyOfRange(wordArray, 1, wordArray.length));
```
Comment (okkhoy) [inline src/main/java/Parser.java:41]: can SLAP this out; see the example from previous lecture.

File: src/main/java/Storage.java [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+        } catch (IOException error) {
+            System.out.println(error);
+        }
+    }
+
+    /**
+     * Loads save file using file path saved in <code>saveFilePath</code> attribute.
+     * Stores all the commands as a String in an array.
+     */
+    public void loadSave() {
```
Comment (okkhoy) [inline src/main/java/Storage.java:36]: the name can be better. reason: does `loadSave` mean load (read from file) or save (write to file)?

File: src/main/java/Storage.java [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+        }
+    }
+
+    /**
+     * Saves all tasks into a text file located in path
+     * stated by attribute <code>saveFilePath</code>.
+     * A new file is created if it a save file does not exist.
+     * @param taskList list of tasks
+     */
+    public void createSave(TaskList taskList) {
```
Comment (okkhoy) [inline src/main/java/Storage.java:60]: same comment as above

File: src/main/java/Storage.java [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+            File saveFile = new File(this.saveFilePath);
+            FileWriter saveFileWriter = new FileWriter(saveFile, true);
+
+            clearPreviousSave(saveFile);
+            
+            ArrayList<Task> list = taskList.getList();
+            for (Task task: list) {
+                switch(task.getType()) {
+                case "T":
+                    saveFileWriter.write("todo " + task.getDescription() + System.lineSeparator());
```
Comment (okkhoy) [inline src/main/java/Storage.java:71]: you do have a toString() method, right? can try to us that here.

File: src/main/java/Storage.java [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+                }
+            }
+            saveFileWriter.close();
+        } catch (IOException error) {
+            System.out.println(error);
+        }
+    }
+
+    /**
+     * @return ArrayList of String containing all the commands from save file
```
Comment (okkhoy) [inline src/main/java/Storage.java:93]: better: Returns the ArrayList of string ...

**Review by okkhoy — COMMENTED** [review]
In general, code is well written. Try to organize the code as per A-Packages for better code management. Pay attention to (some of the) method names.
